From 7480cc0290137d4671e6b521ba0264fd747df0e6 Mon Sep 17 00:00:00 2001
From: Timon <2370185+timonsku@users.noreply.github.com>
Date: Thu, 7 Dec 2023 22:31:52 +0100
Subject: [PATCH] configs: rpi: Compile TSC2007 as module

Compile the TSC2007 touchscreen driver as a module
and enable the optional IIO module.

Signed-off-by: Timon Skerutsch <kernel@diodes-delight.com>
---
 arch/arm/configs/bcm2709_defconfig   | 2 ++
 arch/arm/configs/bcm2711_defconfig   | 2 ++
 arch/arm/configs/bcmrpi_defconfig    | 2 ++
 arch/arm64/configs/bcm2711_defconfig | 2 ++
 arch/arm64/configs/bcm2712_defconfig | 2 ++
 arch/arm64/configs/bcmrpi3_defconfig | 2 ++
 6 files changed, 12 insertions(+)

--- a/arch/arm/configs/bcm2709_defconfig
+++ b/arch/arm/configs/bcm2709_defconfig
@@ -625,6 +625,8 @@ CONFIG_TOUCHSCREEN_ILI210X=m
 CONFIG_TOUCHSCREEN_EDT_FT5X06=m
 CONFIG_TOUCHSCREEN_RASPBERRYPI_FW=m
 CONFIG_TOUCHSCREEN_USB_COMPOSITE=m
+CONFIG_TOUCHSCREEN_TSC2007=m
+CONFIG_TOUCHSCREEN_TSC2007_IIO=y
 CONFIG_TOUCHSCREEN_STMPE=m
 CONFIG_TOUCHSCREEN_IQS5XX=m
 CONFIG_INPUT_MISC=y
--- a/arch/arm/configs/bcm2711_defconfig
+++ b/arch/arm/configs/bcm2711_defconfig
@@ -639,6 +639,8 @@ CONFIG_TOUCHSCREEN_ILI210X=m
 CONFIG_TOUCHSCREEN_EDT_FT5X06=m
 CONFIG_TOUCHSCREEN_RASPBERRYPI_FW=m
 CONFIG_TOUCHSCREEN_USB_COMPOSITE=m
+CONFIG_TOUCHSCREEN_TSC2007=m
+CONFIG_TOUCHSCREEN_TSC2007_IIO=y
 CONFIG_TOUCHSCREEN_STMPE=m
 CONFIG_TOUCHSCREEN_IQS5XX=m
 CONFIG_INPUT_MISC=y
--- a/arch/arm/configs/bcmrpi_defconfig
+++ b/arch/arm/configs/bcmrpi_defconfig
@@ -620,6 +620,8 @@ CONFIG_TOUCHSCREEN_ILI210X=m
 CONFIG_TOUCHSCREEN_EDT_FT5X06=m
 CONFIG_TOUCHSCREEN_RASPBERRYPI_FW=m
 CONFIG_TOUCHSCREEN_USB_COMPOSITE=m
+CONFIG_TOUCHSCREEN_TSC2007=m
+CONFIG_TOUCHSCREEN_TSC2007_IIO=y
 CONFIG_TOUCHSCREEN_STMPE=m
 CONFIG_TOUCHSCREEN_IQS5XX=m
 CONFIG_INPUT_MISC=y
--- a/arch/arm64/configs/bcm2711_defconfig
+++ b/arch/arm64/configs/bcm2711_defconfig
@@ -656,6 +656,8 @@ CONFIG_TOUCHSCREEN_ILI210X=m
 CONFIG_TOUCHSCREEN_EDT_FT5X06=m
 CONFIG_TOUCHSCREEN_RASPBERRYPI_FW=m
 CONFIG_TOUCHSCREEN_USB_COMPOSITE=m
+CONFIG_TOUCHSCREEN_TSC2007=m
+CONFIG_TOUCHSCREEN_TSC2007_IIO=y
 CONFIG_TOUCHSCREEN_STMPE=m
 CONFIG_TOUCHSCREEN_IQS5XX=m
 CONFIG_INPUT_MISC=y
--- a/arch/arm64/configs/bcm2712_defconfig
+++ b/arch/arm64/configs/bcm2712_defconfig
@@ -659,6 +659,8 @@ CONFIG_TOUCHSCREEN_ILI210X=m
 CONFIG_TOUCHSCREEN_EDT_FT5X06=m
 CONFIG_TOUCHSCREEN_RASPBERRYPI_FW=m
 CONFIG_TOUCHSCREEN_USB_COMPOSITE=m
+CONFIG_TOUCHSCREEN_TSC2007=m
+CONFIG_TOUCHSCREEN_TSC2007_IIO=y
 CONFIG_TOUCHSCREEN_STMPE=m
 CONFIG_TOUCHSCREEN_IQS5XX=m
 CONFIG_INPUT_MISC=y
--- a/arch/arm64/configs/bcmrpi3_defconfig
+++ b/arch/arm64/configs/bcmrpi3_defconfig
@@ -623,6 +623,8 @@ CONFIG_TOUCHSCREEN_EKTF2127=m
 CONFIG_TOUCHSCREEN_EDT_FT5X06=m
 CONFIG_TOUCHSCREEN_RASPBERRYPI_FW=m
 CONFIG_TOUCHSCREEN_USB_COMPOSITE=m
+CONFIG_TOUCHSCREEN_TSC2007=m
+CONFIG_TOUCHSCREEN_TSC2007_IIO=y
 CONFIG_TOUCHSCREEN_STMPE=m
 CONFIG_TOUCHSCREEN_IQS5XX=m
 CONFIG_INPUT_MISC=y
